adal/src/main/java/com/microsoft/aad/adal/AuthenticationResult.java

```diff
@@ -109,6 +112,21 @@
         mIdToken = idToken;
     }
 
+    AuthenticationResult(String accessToken, String refreshToken, Date expires, boolean isBroad,
+                         UserInfo userInfo, String tenantId, String idToken, Date extendedExpires) {
+        mCode = null;
+        mAccessToken = accessToken;
+        mRefreshToken = refreshToken;
+        mExpiresOn = expires;
+        mIsMultiResourceRefreshToken = isBroad;
+        mStatus = AuthenticationStatus.Succeeded;
+        mUserInfo = userInfo;
+        mTenantId = tenantId;
+        mIdToken = idToken;
+        if (extendedExpires != null) {
```

Comment:
i don't see this if check adding any value. is there a reason why we prefer unassigned over null value?